### PR TITLE
[libc] Fix standalone cross compiling build for the GPU

### PR DESCRIPTION
**Pull request by jhuber6**
Summary:
This patch fixes some issues with building the GPU target manually
without the runtimes bootstrapping build. This fixes the install
directory and sets the default namespace to something more sensible if
not set. Also I got rid of the check on `-mcpu=native`. it was a neat
trick, but CMake in its INFINITE wisdom does not allow you to set link
flags on the compiler flag check. So I just went with the old tool
usage.

### CODE REVIEW

File: libc/CMakeLists.txt [hunk 1]
```diff
@@ -39,7 +39,11 @@ set(LIBC_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR})
 set(LIBC_ENABLE_USE_BY_CLANG OFF CACHE BOOL "Whether or not to place libc in a build directory findable by a just built clang")
 
 # Defining a global namespace to enclose all libc functions.
-set(LIBC_NAMESPACE "__llvm_libc_${LLVM_VERSION_MAJOR}_${LLVM_VERSION_MINOR}_${LLVM_VERSION_PATCH}_${LLVM_VERSION_SUFFIX}"
+set(default_namespace "__llvm_libc_")
```
Comment (lntue) [inline libc/CMakeLists.txt:42]: `__llvm_libc`?

Comment (jhuber6) [reply libc/CMakeLists.txt:42]: Forgot to delete that.
```suggestion
set(default_namespace "__llvm_libc")
```

**Review by lntue — APPROVED** [review]

File: libc/CMakeLists.txt [hunk 2]
```diff
@@ -281,7 +291,7 @@ if(LLVM_LIBC_ENABLE_LINTING)
           result : '${CLANG_TIDY_RESULT}'
           ")
       endif()
-      string(REGEX MATCH "[0-9]+" CLANG_TIDY_VERSION "${CLANG_TIDY_OUTPUT}")
+      string(REGEX MATCH "+" CLANG_TIDY_VERSION "${CLANG_TIDY_OUTPUT}")
```
[the remark this answers is not in the document]
Comment (jhuber6) [reply libc/CMakeLists.txt:294]: Wait how did get there.
```suggestion
      string(REGEX MATCH "[0-9]+" CLANG_TIDY_VERSION "${CLANG_TIDY_OUTPUT}")
```

File: libc/CMakeLists.txt [hunk 3]
```diff
@@ -39,7 +39,11 @@ set(LIBC_BUILD_DIR ${CMAKE_CURRENT_BINARY_DIR})
 set(LIBC_ENABLE_USE_BY_CLANG OFF CACHE BOOL "Whether or not to place libc in a build directory findable by a just built clang")
 
 # Defining a global namespace to enclose all libc functions.
-set(LIBC_NAMESPACE "__llvm_libc_${LLVM_VERSION_MAJOR}_${LLVM_VERSION_MINOR}_${LLVM_VERSION_PATCH}_${LLVM_VERSION_SUFFIX}"
+set(default_namespace "__llvm_libc")
```
Comment (michaelrj-google) [inline libc/CMakeLists.txt:42]: is this change related to the rest of this PR? It seems like an odd inclusion

Comment (nickdesaulniers) [reply libc/CMakeLists.txt:42]: Why would `LLVM_VERSION_MAJOR` not be set? Does this happen for the runtimes build? (if so, cc @petrhosek )

Comment (jhuber6) [reply libc/CMakeLists.txt:42]: This happens if you build it directly out of the `runtimes/` directory. We specifically build everything else independent from LLVM if possible so I don't think this is a surprising change to have a fallback if the user is building out-of-tree or manually.

**Review by michaelrj-google — APPROVED** [review]

File: libc/CMakeLists.txt (hunk 3 above)
Comment (nickdesaulniers) [reply libc/CMakeLists.txt:42]: I wonder if building via the runtimes directory SHOULD set `LLVM_VERSION_MAJOR`? Otherwise libcxx, compiler-rt, and now llvmlibc wont be able to key off this value.

The non `LIBC_NAMESPACE` changes LGTM, if you want to split this PR in two, I'd accept the second half.  For the first, I suspect we may want to fix the runtimes CMake itself.

Comment (jhuber6) [reply libc/CMakeLists.txt:42]: We already want to support out-of-tree builds, I don't see the issue with setting a sane default for those cases.

Comment (nickdesaulniers) [reply libc/CMakeLists.txt:42]: > We already want to support out-of-tree builds

What do you mean? Folks still need to use llvm/ or runtimes/ if they use cmake. If they're not using cmake, then setting a default namespace for cmake is irrelevant.

Comment (jhuber6) [reply libc/CMakeLists.txt:42]: All this does is change `__llvm_libc_____` into `__llvm_libc` is this a blocking issue or can I merge the patch as-is.
